pytorch_lightning/loops/base.py

```diff
@@ -217,7 +217,7 @@ def _load_from_state_dict(
             if isinstance(v, BaseProgress):
                 v.load_state_dict(state_dict[key])
                 if restart_progress:
-                    apply_to_collection(v, Progress, lambda p: p.current.reset_on_restart())
+                    apply_to_collection(v, BaseProgress, lambda p: p.reset_on_restart())
```

Comment:
Why is this necessary?

Not reseting `p.total` is done on purpose.

Why do we need this `restart_progress` flag and this code here instead of relying on

```python
def reset():
    if self.restarting:
         reset_something()
```

Reply:
`OptimizationProgress` was not properly resetted. It took me some to find this. I think it is safer to always call the BaseProgress and rely on the reset_on_restart to not reset progress.

Reply:
IMO, I prefer the loop to take care of sanitising progress.